### PR TITLE
Enable C# semantic tests on CoreClr

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @roslyn-compiler for review

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-compiler for review

File: build/scripts/tests.sh [hunk 1]
```diff
@@ -69,7 +69,8 @@ do
     echo Running "${runtime} ${file_name[@]}"
     if [[ "${runtime}" == "dotnet" ]]; then
         runner="dotnet exec --depsfile ${deps_json} --runtimeconfig ${runtimeconfig_json}"
-        if [[ "${file_name[@]}" == *'Roslyn.Compilers.CSharp.Emit.UnitTests.dll' ]] || [[ "${file_name[@]}" == *'Roslyn.Compilers.VisualBasic.Symbol.UnitTests.dll' ]]
+        if [[ "${file_name[@]}" == *'Roslyn.Compilers.CSharp.Emit.UnitTests.dll' ]] || [[ "${file_name[@]}" == *'Roslyn.Compilers.VisualBasic.Symbol.UnitTests.dll' ]] || [[ "${file_name[@]}" == *'Roslyn.Compilers.CSharp.Semantic.UnitTests.dll' ]]
```
Comment (333fred) [inline build/scripts/tests.sh]: Nit: consider breaking up long line. It looks fine on my 4k monitor, but would not on another monitor.

Comment (jaredpar) [reply build/scripts/tests.sh]: Do you know what the newline rules are for bash script? I tried looking it up but couldn't find anything. If I knew the rule I'd break it up accordingly

File: src/Compilers/CSharp/Test/Semantic/Properties/launchSettings.json [hunk 2]
```diff
@@ -1,12 +1,13 @@
 {
   "profiles": {
     "xUnit.net Console": {
-      "executablePath": "$(XUnitPath)",
-      "commandLineArgs": "$(XUnitArguments)"
+      "commandName": "Executable",
+      "executablePath": "C:\\Program Files\\dotnet\\dotnet.exe",
+      "commandLineArgs": "exec --depsfile E:\\code\\roslyn\\Binaries\\Debug\\UnitTests\\CSharpCompilerSemanticTest\\netcoreapp2.0\\Roslyn.Compilers.CSharp.Semantic.UnitTests.deps.json --runtimeconfig E:\\code\\roslyn\\Binaries\\Debug\\UnitTests\\CSharpCompilerSemanticTest\\netcoreapp2.0\\Roslyn.Compilers.CSharp.Semantic.UnitTests.runtimeconfig.json e:\\nuget\\xunit.runner.console\\2.3.1\\tools\\netcoreapp2.0\\xunit.console.dll E:\\code\\roslyn\\Binaries\\Debug\\UnitTests\\CSharpCompilerSemanticTest\\netcoreapp2.0\\Roslyn.Compilers.CSharp.Semantic.UnitTests.dll -method Microsoft.CodeAnalysis.CSharp.UnitTests.OutVarTests.Query_01"
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Properties/launchSettings.json]: No

File: build/scripts/tests.sh (hunk 1 above)
Comment (333fred) [reply build/scripts/tests.sh]: iirc either backslashes or evaluate conditions ahead of time into a variable and check the variable

---
In reply to: [196973329](https://github.com/dotnet/roslyn/pull/28030#discussion_r196973329) [](ancestors = 196973329)

File: src/Compilers/CSharp/Test/Semantic/Properties/launchSettings.json (hunk 2 above)
Comment (jaredpar) [reply src/Compilers/CSharp/Test/Semantic/Properties/launchSettings.json]: The change to not have these settings in a .user file is so frustrating. Every single time you have to debug you break things. 😦

**Comment by jaredpar** [comment]
Responded to PR feedback

File: build/scripts/tests.sh (hunk 1 above)
Comment (jaredpar) [reply build/scripts/tests.sh]: Tried backslash and it ended up giving syntax errors. Will address this in a future PR when I get back to making these all run on Linux and I have bash readily available to rapidly try this out.

File: src/Compilers/CSharp/Test/Semantic/Properties/launchSettings.json [hunk 3]
```diff
@@ -9,4 +9,4 @@
       "commandLineArgs": "$(XUnitWpfArguments)"
     }
   }
-}
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Properties/launchSettings.json]: Consider reverting

**Pull request merged**